pkg/operator/controller/controller_dns_daemonset_test.go

```diff
@@ -166,6 +166,22 @@ func TestDaemonsetConfigChanged(t *testing.T) {
 			},
 			expect: true,
 		},
+		{
+			description: "if the config-volume default mode value is defaulted",
+			mutate: func(daemonset *appsv1.DaemonSet) {
+				newVal := volumeDefaultMode
+				daemonset.Spec.Template.Spec.Volumes[1].Secret.DefaultMode = &newVal
+			},
+			expect: false,
+		},
+		{
+			description: "if the config-volume default mode value changes",
+			mutate: func(daemonset *appsv1.DaemonSet) {
+				newVal := int32(0)
+				daemonset.Spec.Template.Spec.Volumes[1].Secret.DefaultMode = &newVal
+			},
+			expect: true,
```

Comment:
These 2 test cases cover the `cmpSecretVolumeSource(...)` function. Perhaps there should be 2 similar test cases for `cmpConfigMapVolumeSource(...)`?

Reply:
Added.  I also noticed that the index was wrong (it should be 0 rather than 1 for the `config-volume` volume), so I fixed the index on the existing test and added a host-path volume to make the list of volumes consistent with the actual manifest.  Thanks!